modules/exploits/unix/webapp/drupal_sa_core_2019_003.rb

```diff
@@ -0,0 +1,218 @@
+      'Description'    => %q{
+        This module exploits a Drupal RESTful Web Services RCE.
+
+        Drupal < 8.5.11, and < 8.6.10 are vulnerable.
+      },
+      'Author'         => [
+        'Samuel Mortenson', # Vulnerability discovery
+        'Charles Fol',      # Proof of concept (Drupal 8.x)
+        'Rotem Reiss',      # Metasploit module
+        'wvu'               # Author of the drupal_drupalgeddon2 Metasploit module which was used as the base for this module
+      ],
+      'References'     => [
+        ['CVE', '2019-6340'],
+        ['URL', 'https://www.drupal.org/sa-core-2019-003'],
+        ['URL', 'https://www.ambionics.io/blog/drupal8-rce']
+      ],
+      'DisclosureDate' => '2019-02-20',
+      'License'        => MSF_LICENSE,
+      'Platform'       => ['php', 'unix', 'linux'],
+      'Arch'           => [ARCH_PHP, ARCH_CMD, ARCH_X86, ARCH_X64],
+      'Privileged'     => false,
+      'Payload'        => {'BadChars' => '&>\''},
+      'Targets'        => [
+        #
+        # Automatic targets (PHP, cmd/unix, native)
+        #
+        ['Automatic (PHP In-Memory)',
+          'Platform'   => 'php',
+          'Arch'       => ARCH_PHP,
+          'Type'       => :php_memory
+        ],
+        ['Automatic (PHP Dropper)',
+          'Platform'   => 'php',
+          'Arch'       => ARCH_PHP,
+          'Type'       => :php_dropper
+        ],
+        ['Automatic (Unix In-Memory)',
+          'Platform'   => 'unix',
+          'Arch'       => ARCH_CMD,
+          'Type'       => :unix_memory
+        ],
+        ['Automatic (Linux Dropper)',
+          'Platform'   => 'linux',
+          'Arch'       => [ARCH_X86, ARCH_X64],
+          'Type'       => :linux_dropper
+        ],
+        #
+        # Drupal 8.x targets (PHP, cmd/unix, native)
+        #
+        ['Drupal 8.x (PHP In-Memory)',
+          'Platform'   => 'php',
+          'Arch'       => ARCH_PHP,
+          'Version'    => Gem::Version.new('8'),
+          'Type'       => :php_memory
+        ],
+        ['Drupal 8.x (PHP Dropper)',
+          'Platform'   => 'php',
+          'Arch'       => ARCH_PHP,
+          'Version'    => Gem::Version.new('8'),
+          'Type'       => :php_dropper
+        ],
+        ['Drupal 8.x (Unix In-Memory)',
+          'Platform'   => 'unix',
+          'Arch'       => ARCH_CMD,
+          'Version'    => Gem::Version.new('8'),
+          'Type'       => :unix_memory
+        ],
+        ['Drupal 8.x (Linux Dropper)',
+          'Platform'   => 'linux',
+          'Arch'       => [ARCH_X86, ARCH_X64],
+          'Version'    => Gem::Version.new('8'),
+          'Type'       => :linux_dropper
+        ]
+      ],
+      'DefaultTarget'  => 0, # Automatic (PHP In-Memory)
+      'Notes'          => {'AKA' => ['SA-CORE-2019-003', 'CVE-2019-6340']}
+    ))
+
+    register_options([
+      OptBool.new('DUMP_OUTPUT', [false, 'Dump HTTP response body', false])
```

Comment:
Some of this is still old code, hence why copying modules kinda sucks.

Reply:
Can you assist me with that? When I removed some of the targets the payload just stopped working

Reply:
Would you be okay with my rewriting a bit of this PR and then sending you a PR to review and potentially merge? I'm happy to do some heavy lifting on this.

Reply:
That's cool, I already spent on it much more time than planned..

Reply:
That's cool, I already spent on it much more time than planned..

Reply:
No worries. I'll tackle it in the morning!

Reply:
Any updates? Do you need my help with that?

Reply:
Getting around to it. A lot on my plate. You'll see it completed this week!